src/cv2pdb.cpp

```diff
@@ -17,8 +17,8 @@
 
 static const int typePrefix = 4;
 
-CV2PDB::CV2PDB(PEImage& image, DebugLevel debug_)
-: img(image), pdb(0), dbi(0), tpi(0), ipi(0), libraries(0), rsds(0), rsdsLen(0), modules(0), globmod(0)
+CV2PDB::CV2PDB(PEImage& image, PEImage* imageDWARF, DebugLevel debug_)
+: img(image), imgDWARF(imageDWARF ? imageDWARF : &image), pdb(0), dbi(0), tpi(0), ipi(0), libraries(0), rsds(0), rsdsLen(0), modules(0), globmod(0)

```

Comment:
I think it is a bit confusing that `imgDWARF` is also used for the CodeView part. I haven't checked how often `img` is still used now, but couldn't it be initialized with `img(imageDWARF ? *imageDWARF : image)` instead, and `image ` be kept in some `exeImage` reference/pointer?

Reply:
`imgDWARF` will always be NULL for CodeView.

If exe image has either CV entries or DWARF entries, it will be assigned to `img` and `imgDWARF` will be NULL.  This logic remains as in the original code.

If there is a debug link, either specified by the switch or present in the exe image, code will attempt to load the linked DWARF image.  In this case, `img` will be the exe image and `imgDWARF` will be the linked DWARF image.

Otherwise a separate .dbg is loaded (CodeView) and assigned to `img` and `imgDWARF` will be NULL.

Reply:
In CV2PDB initialization, `imageDWARF` will refer to `img` if the pointer was NULL.

Perhaps renaming imageDWARF is the correct choice because it can point to `img` which in turn can be either exe image or CV image.

Reply:
`img` is what code modifies while `imgDWARF` is a "read-only" image.  Just trying to come up with a good name for it.

Reply:
I renamed `imgDWARF` member to `cimgDebugInfo` and made it const

Reply:
Ok, img is still used for other purposes. `cimgDebugInfo` is fine with me, though a bit long. Maybe just `imgDI` or `imgDbg` with a commit at the declaration site?

Reply:
> with a commit at the declaration site

Don't understand the lingo.

Reply:
Done

Reply:
> Don't understand the lingo. Did you mean "comment"?

Oops. Indeed that's what I meant ;)